### ISSUE TITLE
Contribution guide does not mention our "no merge commits" policy

### ISSUE THREAD

**Issue by oli-obk** [rust-lang/rust#59233]
We expect everyone to rebase instead of merge and we also try to have fixup commits squashed into previous related commits. I think we should document this policy, since it keeps suprising newcomers.

**Comment by stepnivlk** [comment]
@oli-obk I can take care of this one as it's quite a fresh experience :).

**Comment by lzutao** [comment]
I think we should only rebase when resolving merge conflict, or the review process has completed
and reviewer request a rebasing.
Otherwise I prefer adding new commits instead for easier reviewing.

**Comment by bovinebuddha** [comment]
@oli-obk

I agree with @lzutao that this policy is sound for actually merging the PR, but it might cause confusion for reviews. I've had a PR lying around for awhile, and when there is merge conflicts the 'waiting-on-author' label gets added so noone reviews it, but when I do the rebasing to get back my 'waiting-on-review' it seems as if Github sometimes loses context of previous reviews, which can be quite confusing...

**Comment by oli-obk** [comment]
for the record: I personally want no rebasing and just merging, I'm just stating what the policy is.

**Comment by lzutao** [comment]
Before we close this issue. Could we talk a bit about [closing issues using keyword](https://help.github.com/en/articles/closing-issues-using-keywords)?

> For example, to close an issue numbered 123, you could use the phrase "Closes #123" or "Closes: #123" in your pull request description or commit message. Once the branch is merged into the default branch, the issue will close.

I think we should encourage contributors close issues by keywords in PR descriptions only.
Because otherwise the commit could be rebased many times, and each time it creates a reference to
the want-to-close issue. Also in the referenced issue, the commit number of the new commit
that could close that issue is not really informative. The PR number itself is more informative
and concise.
